app/services/check_participant_eligibility.rb

```diff
@@ -9,6 +9,21 @@ def initialize(trn:)
 
   def call
     ineligible_record = ECFIneligibleParticipant.find_by(trn: trn)
-    return ineligible_record.reason.to_sym if ineligible_record.present?
+
+    # logic is reversed in the data snapshot from TRA (:previous_induction)
+    # so if your TRN is present, you're valid
+    if ineligible_record.present?
+      case ineligible_record.reason.to_sym
+      when :previous_induction
+        # this is reversed and indicates that the TRN is eligible
+        nil
+      when :previous_induction_and_participation
+        :previous_participation
+      when :previous_participation
+        :previous_induction_and_participation
+      end
+    else
+      :previous_induction
+    end
```

Comment:
I'm glad this won't be around for long. Not your fault, but it's horrible 😛

Reply:
Completely agree!